frontend/src/routes/signup/logic.ts

```diff
@@ -0,0 +1,43 @@
+import { zodResolver } from '@hookform/resolvers/zod';
+import { useForm } from 'react-hook-form';
+import { z } from 'zod';
+
+export const signUpSchema = z
+  .object({
+    email: z.string().email(),
+    username: z.string().min(2).max(50),
+    firstName: z.string().min(2).max(50),
```

Comment:
![image](https://github.com/user-attachments/assets/8cc6612d-e72e-4804-8d57-b206795a0723)
Can we add custom error messages for invalid inputs for a better user experience. Default error message is **String** must contain....
Users may not know what strings are